File: chpt1/section_1_3/problem_1_3_3.m
% 1.3.3 Start with y(0) = $5000. If this grows by dy/dt = 0.02y until
% t = 5 and then jumps to a = 0.04 per year until t = 10. What is the
% account balance at t = 10?
t1 = 0.0:0.01:5.0;
t2 = 5.0:0.01:10.0;

y1=5000*exp(0.02*t1);
y2=5000*exp((0.04*t2)-0.1);

plot(t1,y1,t2,y2);
legend("y1", "y2");
ylim([5000 7000]);
xlim([0 11])
grid on;
title("1.3.3 Plot of y1(t)=5000e^{0.02t} and y2(t)=5000e^{(0.04*t2)-0.1}");
xlabel("time");
ylabel("y");

File: chpt1/section_1_3/problem_1_3_3.py
import numpy as np
import matplotlib.pyplot as plt

t1 = np.linspace(0, 5, 400)  # 400 points from 0 to 1
t2 = np.linspace(5, 10, 400)  # 400 points from 1 to 2
y1 = 5000*np.exp(0.2*t1)  # Compute e^{t} for each t
y2 = 5000*np.exp((0.4*t2)-1)

plt.figure(figsize=(8, 4))
plt.plot(t1, y1, label='$y1 = 5000*e^{0.2*t}$')
plt.plot(t2, y2, label='$y2 = 5000*e^{(0.4*t)-1}$')

# Adding title and labels
plt.title('Graph of $y1 = 5000*e^{0.2*t}$ and $y2 = 5000*e^{(0.4*t)-1}$')
plt.xlabel('t')
plt.ylabel('y')

# Add a grid and legend
plt.grid(True)
plt.legend()

# Show the plot
plt.show()